File: src/main/kotlin/info/hildegynoid/cue/converter/entity/CueSheet.kt
package info.hildegynoid.cue.converter.entity

import info.hildegynoid.cue.converter.loader.EOL
import info.hildegynoid.cue.converter.loader.escape

data class CueSheet(
    var title: String = "",
    var performer: String = "",
    var remarks: MutableMap<String, String> = mutableMapOf(),
    var files: MutableList<CueFileInfo> = mutableListOf()
) {
    override fun toString(): String {
        return remarks.map {
            "REM ${it.key} \"${it.value.escape()}\"" + EOL
        }.joinToString("") +
            "TITLE \"${title.escape()}\"" + EOL +
            "PERFORMER \"${performer.escape()}\"" + EOL +
            files.joinToString("") { it.toString() }
    }
}

File: src/cue/entity/CueSheet.ts
import { CueFileInfo } from './CueFileInfo';
import { EOL, escapeString } from '../Utils';

export class CueSheet {
  title: string = '';

  performer: string = '';

  remarks: Map<string, string> = new Map();

  files: Array<CueFileInfo> = [];

  getTracks(): Array<string> {
    let list = Array<string>();
    this.files.forEach((file) => {
      list = list.concat(file.getTrackList());
    });
    return list;
  }

  toString = (): string => {
    let result = '';

    this.remarks.forEach((value, key) => {
      result += `REM ${key} "${escapeString(value)}"${EOL}`;
    });

    result += `TITLE "${escapeString(this.title)}"${EOL}`;
    result += `PERFORMER "${escapeString(this.performer)}"${EOL}`;
    result += this.files.join('');
    return result;
  };
}
